samcli/lib/package/ecr_uploader.py

```diff
@@ -83,6 +90,40 @@ def upload(self, image, resource_name):
 
         return f"{repository}:{_tag}"
 
+    def delete_artifact(self, image_uri: str, resource_id: str, property_name: str):
+        """
+        Delete the given ECR image by extracting the repository and image_tag from
+        image_uri
+
+        :param image_uri: image_uri of the image to be deleted
+        :param resource_id: id of the resource for which the image is deleted
+        :param property_name: provided property_name for the resource
+        """
+        try:
+            repo_image_tag = image_uri.split("/")[1].split(":")
+            repository = repo_image_tag[0]
+            image_tag = repo_image_tag[1]
```

Comment:
is it possible that the image tag is in Format `repository:tag` and not in format `registry/repository:tag`. I think also the tag is optional, which means that the tag value is `latest`.

I think you need also to guard the code against wrong formats, I know that this code depends on the template you downloaded from CFN which means that it will be always a correct format, but think from code reusability point of view, so this method should be self contained, and does not depend on the caller.